src/utils.ts

```diff
@@ -0,0 +1,165 @@
+import type {
+  AsyncFn,
+  HtmlEscapes,
+  ISbResult,
+  ISbStoriesParams,
+} from './interfaces';
+
+// TODO: Revise this type, is it needed?
+interface ISbParams extends ISbStoriesParams {
+  [key: string]: any;
+}
+
+type ArrayFn = (...args: any) => void;
+type FlatMapFn = (...args: any) => [] | any;
+type RangeFn = (...args: any) => [];
+
+/**
+ * Checks if a URL is a CDN URL
+ * @param url - The URL to check
+ * @returns boolean indicating if the URL is a CDN URL
+ */
+export const isCDNUrl = (url = ''): boolean => url.includes('/cdn/');
+
+/**
+ * Gets pagination options for the API request
+ * @param options - The base options
+ * @param perPage - Number of items per page
+ * @param page - Current page number
+ * @returns Object with pagination options
+ */
+export const getOptionsPage = (
+  options: ISbStoriesParams,
+  perPage = 25,
+  page = 1,
+) => ({
+  ...options,
+  per_page: perPage,
+  page,
+});
+
+/**
+ * Creates a promise that resolves after the specified milliseconds
+ * @param ms - Milliseconds to delay
+ * @returns Promise that resolves after the delay
+ */
+export const delay = (ms: number): Promise<void> =>
+  new Promise(res => setTimeout(res, ms));
+
+/**
+ * Creates an array of specified length using a mapping function
+ * @param length - Length of the array
+ * @param func - Mapping function
+ * @returns Array of specified length
+ */
+export const arrayFrom = (length = 0, func: ArrayFn) =>
+  Array.from({ length }, func);
+
+/**
+ * Creates an array of numbers in the specified range
+ * @param start - Start of the range
+ * @param end - End of the range
+ * @returns Array of numbers in the range
+ */
+export const range = (start = 0, end = start): Array<any> => {
+  const length = Math.abs(end - start) || 0;
+  const step = start < end ? 1 : -1;
+  return arrayFrom(length, (_, i: number) => i * step + start);
+};
+
+/**
+ * Maps an array asynchronously
+ * @param arr - Array to map
+ * @param func - Async mapping function
+ * @returns Promise resolving to mapped array
+ */
+export const asyncMap = async (arr: RangeFn[], func: AsyncFn) =>
+  Promise.all(arr.map(func));
+
+/**
+ * Flattens an array using a mapping function
+ * @param arr - Array to flatten
+ * @param func - Mapping function
+ * @returns Flattened array
+ */
+export const flatMap = (arr: ISbResult[] = [], func: FlatMapFn) =>
+  arr.map(func).reduce((xs, ys) => [...xs, ...ys], []);
```

Comment:
[`Array.prototype.flatMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) has been supported in Node _and_ in browser for a very very long time, so this abstraction is redundant.

Reply:
@ryami333 while I can agree with this and your previous comment, those functions are there for historical reasons and we'll change them in a future major release ;)

Reply:
Hi @ryami333, yes, right now we are just porting things out away from classes, there are going to be substantial changes to the package in the next major versions so this utils would likely change.